torch/csrc/jit/import.cpp

```diff
@@ -42,6 +42,12 @@ void postSetStateValidate(const IValue& v) {
     // Verify that all the non-optional attributes have been initialized
     // TODO: Issue #20497
     if (attrType->kind() != TypeKind::OptionalType) {
+      if (slot.isNone() && attrName == "training") {
```

Comment:
I think this counts as another special case. I don't think it's correct to assume that anyone who defines a `__setstate__` will want their module to default to `eval()` (although it's probably true in most circumstances). 

We should just require that they initialize `self.training` in `__setstate__`, like any other attribute of the module. It's a little annoying, but more consistent overall.

Reply:
You'll have to fix up places in our standard lib which do this now. As above, https://github.com/pytorch/pytorch/pull/26666 will tell you where you need to fix up, because I made the same change in that PR.